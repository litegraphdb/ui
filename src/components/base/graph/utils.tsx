import { NodeData } from '@/lib/graph/types';
import { defaultNodeColor, nodeLightColorByType, universalNodeColors } from './constant';

const maxLegends = 16;

export const getLegendsForNodes = (nodes: NodeData[]) => {
  const priorityLegends = new Map<string, { legend: string; color: string }>();
  const customLegends = new Map<string, { legend: string; color: string }>();

  // Separate legends into priority (nodeLightColorByType) and custom
  nodes.forEach((node) => {
    if (Object.keys(nodeLightColorByType).includes(node.type)) {
      priorityLegends.set(node.type, { legend: node.type, color: nodeLightColorByType[node.type] });
    } else {
      // Only add if not already present to avoid changing colors for same legend
      if (!customLegends.has(node.type)) {
        customLegends.set(node.type, {
          legend: node.type,
          color: universalNodeColors[customLegends.size % universalNodeColors.length],
        });
      }
    }
  });

  const priorityLegendsSize = priorityLegends.size;
  // If there are nodes with unknown types, add "Unknown" to priority legends
  const isMoreThanMaxLegends = customLegends.size + priorityLegendsSize > maxLegends;

  // Convert to arrays for easier manipulation
  const priorityArray = Array.from(priorityLegends.values());
  const customArray = Array.from(customLegends.values()).slice(
    0,
    maxLegends - (priorityArray.length + (isMoreThanMaxLegends ? 1 : 0))
  );

  // Combine priority legends first, then custom legends
  const legends = [
    ...priorityArray,
    ...customArray,
    ...(isMoreThanMaxLegends ? [{ legend: 'Unknown', color: defaultNodeColor }] : []),
  ];
  const mapLegends: Record<string, { legend: string; color: string }> = {};
  legends.forEach((legend) => {
    mapLegends[legend.legend] = legend;
  });
  return mapLegends;
};
