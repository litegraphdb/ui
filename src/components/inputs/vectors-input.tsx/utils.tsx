export const convertVectorsToAPIRecord = (
  vectors?: Array<{ Model: string; Dimensionality: number; Content: string; Vectors: number[] }>
) => {
  return (
    vectors?.map((vector: any) => ({
      ...vector,
      Dimensionality: Number(vector.Dimensionality),
      Vectors: vector.Vectors || [],
    })) || []
  );
};
