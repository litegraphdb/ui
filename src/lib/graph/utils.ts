import type { NodeData, Point } from './types';

export function isPointInNode(point: Point, node: NodeData, radius: number = 10): boolean {
  const dx = point.x - node.x;
  const dy = point.y - node.y;
  return dx * dx + dy * dy <= radius * radius;
}

export function getNodeAtPosition(nodes: NodeData[], point: Point): NodeData | null {
  return nodes.find((node) => isPointInNode(point, node)) || null;
}

export function transformToOptions<T extends { GUID: string; name?: string; Name?: string }>(
  data?: T[] | null,
  labelField: keyof T = 'name' // Field to use for label in options
) {
  return data
    ? data?.map((item) => ({
        value: item.GUID,
        label: (item[labelField] as string) || item.Name || item.GUID,
      }))
    : [];
}
