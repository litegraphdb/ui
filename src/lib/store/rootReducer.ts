import { UnknownAction, combineReducers } from '@reduxjs/toolkit';
import liteGraphReducer from './litegraph/reducer';
import { rtkQueryErrorLogger } from './rtkApiMiddlewear';
import { localStorageKeys, paths } from '@/constants/constant';
import graphReducer from './graph/reducer';
import nodeReducer from './node/reducer';
import edgeReducer from './edge/reducer';
import { LitegraphAction } from './litegraph/actions';
import tenantsReducer from './tenants/reducers';
import tagReducer from './tag/reducer';
import labelReducer from './label/reducer';
import credentialReducer from './credential/reducer';
import userReducer from './user/reducer';
import { tenantLists } from './tenants/actions';
import vectorReducer from './vector/reducer';
import backupReducer from './backup/reducer';
import sdkSlice from './rtk/rtkSdkInstance';
const rootReducer = combineReducers({
  [sdkSlice.reducerPath]: sdkSlice.reducer,
  liteGraph: liteGraphReducer,
  graphsList: graphReducer,
  nodesList: nodeReducer,
  edgesList: edgeReducer,
  tenants: tenantsReducer,
  tagsList: tagReducer,
  vectorsList: vectorReducer,
  labelsList: labelReducer,
  credentialsList: credentialReducer,
  usersList: userReducer,
  tenantsList: tenantsReducer,
  backupsList: backupReducer,
});

export const apiMiddleWares = [rtkQueryErrorLogger, sdkSlice.middleware];

export const handleLogout = (path?: string) => {
  localStorage.removeItem(localStorageKeys.token);
  localStorage.removeItem(localStorageKeys.tenant);
  localStorage.removeItem(localStorageKeys.user);
  localStorage.removeItem(localStorageKeys.adminAccessKey);
  localStorage.removeItem(localStorageKeys.serverUrl);
  window.location.href = path ? path : paths.login;
};

const resettableRootReducer = (
  state: ReturnType<typeof rootReducer> | undefined,
  action: UnknownAction
) => {
  if (action.type === LitegraphAction.LOG_OUT) {
    handleLogout(action.payload as any);
  }
  return rootReducer(state, action);
};

export default resettableRootReducer;
