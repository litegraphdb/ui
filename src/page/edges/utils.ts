import { EdgeType } from '@/lib/store/edge/types';
import { NodeType } from '@/lib/store/node/types';

// Utility to find the name by GUID
export const getNodeNameByGUID = (guid: string, nodesList: NodeType[]) => {
  if (!guid) return 'N/A';
  const node = nodesList.find((n: NodeType) => n.GUID === guid);
  return node ? node.Name : guid;
};

export const transformEdgeDataForTable = (edgesList: EdgeType[], nodesList: NodeType[]) => {
  return (
    edgesList?.map((data: EdgeType) => {
      return {
        ...data,
        FromName: getNodeNameByGUID(data.From, nodesList),
        ToName: getNodeNameByGUID(data.To, nodesList),
        key: data.GUID,
      } as EdgeType;
    }) || []
  );
};
