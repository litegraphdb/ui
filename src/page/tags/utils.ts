import { NodeType } from '@/lib/store/node/types';

import { getNodeNameByGUID } from '../edges/utils';
import { EdgeType } from '@/lib/store/edge/types';
import { TagMetaData } from 'litegraphdb/dist/types/types';
import { TagType } from '@/lib/store/tag/types';

// Utility to find the name by GUID
export const getEdgeNameByGUID = (guid: string, edgesList: EdgeType[]) => {
  if (!guid) return 'N/A';
  const edge = edgesList.find((n: EdgeType) => n.GUID === guid);
  return edge ? edge.Name : guid;
};

export const transformTagsDataForTable = (
  tagsList: TagMetaData[],
  nodesList: NodeType[],
  edgesList: EdgeType[]
) => {
  return (
    tagsList?.map((data: TagMetaData) => {
      return {
        ...data,
        NodeName: getNodeNameByGUID(data.NodeGUID, nodesList),
        EdgeName: getEdgeNameByGUID(data.EdgeGUID, edgesList),
        key: data.GUID,
      } as TagType;
    }) || []
  );
};
