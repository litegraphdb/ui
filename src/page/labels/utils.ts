import { NodeType } from '@/types/types';
import { getNodeNameByGUID } from '../edges/utils';

import { getEdgeNameByGUID } from '../tags/utils';
import { LabelMetadata } from 'litegraphdb/dist/types/types';
import { EdgeType } from '@/types/types';
import { LabelMetadataForTable } from './types';

export const transformLabelsDataForTable = (
  labelsList: LabelMetadata[],
  nodesList: NodeType[],
  edgesList: EdgeType[]
) => {
  return (
    labelsList?.map((data: LabelMetadata) => {
      return {
        ...data,
        NodeName: getNodeNameByGUID(data.NodeGUID, nodesList),
        EdgeName: getEdgeNameByGUID(data.EdgeGUID, edgesList),
        key: data.GUID,
      } as LabelMetadataForTable;
    }) || []
  );
};

export const getNodeAndEdgeGUIDsByLabelList = (labelsList: LabelMetadata[]) => {
  const nodeGUIDs = labelsList.map((label) => label.NodeGUID);
  const edgeGUIDs = labelsList.map((label) => label.EdgeGUID);
  return {
    nodeGUIDs: Array.from(new Set(nodeGUIDs)),
    edgeGUIDs: Array.from(new Set(edgeGUIDs)),
  };
};
