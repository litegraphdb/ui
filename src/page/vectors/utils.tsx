import { NodeType } from '@/types/types';
import { getNodeNameByGUID } from '../edges/utils';
import { EdgeType } from '@/types/types';
import { getEdgeNameByGUID } from '../tags/utils';
import { VectorMetadata } from 'litegraphdb/dist/types/types';

type VectorDisplayData = {
  GUID: string;
  TenantGUID: string;
  GraphGUID: string;
  NodeGUID: string;
  EdgeGUID: string;
  Model: string;
  Dimensionality: number;
  Content: string;
  Vectors: number[];
  CreatedUtc: string;
  LastUpdateUtc: string;
  NodeName: string;
  EdgeName: string;
  key: string;
  vectorString: string;
  vectorCount: number;
};

export const transformVectorsDataForTable = (
  vectorsList: VectorMetadata[],
  nodesList: NodeType[],
  edgesList: EdgeType[] = []
): VectorDisplayData[] => {
  return (
    vectorsList?.map((data) => ({
      ...data,
      NodeName: getNodeNameByGUID(data.NodeGUID, nodesList),
      EdgeName: getEdgeNameByGUID(data.EdgeGUID, edgesList),
      key: data.GUID,
      vectorString: Array.isArray(data.Vectors) ? data.Vectors.join(', ') : '',
      vectorCount: Array.isArray(data.Vectors) ? data.Vectors.length : 0,
    })) || []
  );
};

export const getNodeAndEdgeGUIDsByVectorList = (vectorsList: VectorMetadata[]) => {
  const nodeGUIDs = vectorsList.map((vector) => vector.NodeGUID);
  const edgeGUIDs = vectorsList.map((vector) => vector.EdgeGUID);
  return {
    nodeGUIDs: Array.from(new Set(nodeGUIDs)),
    edgeGUIDs: Array.from(new Set(edgeGUIDs)),
  };
};
