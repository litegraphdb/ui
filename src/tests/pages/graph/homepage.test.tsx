import { screen } from '@testing-library/react';
import '@testing-library/jest-dom';
import { renderWithRedux } from '../../store/utils';
import { createMockStore } from '../../store/mockStore';
import { mockGraphData } from '../mockData';
import { waitFor } from '@testing-library/react';
import DashboardHomePage from '@/app/dashboard/[tenantId]/page';

let container: any;
describe('HomePage with Mock API', () => {
  afterEach(() => {
    container = document.createElement('div');
    document.body.appendChild(container);
  });
  afterEach(() => {
    document.body.removeChild(container);
    container = null;
  });

  const store = createMockStore();

  it('renders the GraphViewer and select one graph', async () => {
    const wrapper = renderWithRedux(
      <DashboardHomePage />,
      {
        ...store,
        graphsList: {
          graphs: mockGraphData,
        },
        liteGraph: {
          ...store.getState().liteGraph,
          selectedGraph: 'e6d4294e-6f49-4d67-8260-5e44c2b077a6',
        },
      } as any,
      true
    );

    waitFor(() => {
      // Verify the select dropdown for graphs
      const graphSelect = screen.getByTestId('litegraph-select');
      expect(graphSelect).toBeInTheDocument();
      expect(graphSelect).toHaveTextContent('Test Demo Graph'); // Adjusted expectation for the mock
    });
    expect(wrapper.container).toMatchSnapshot();
  });
});
