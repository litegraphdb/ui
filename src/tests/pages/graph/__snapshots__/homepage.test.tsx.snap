// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HomePage with Mock API renders the GraphViewer and select one graph 1`] = `
<div>
  <div
    style="position: fixed; z-index: 9999; top: 16px; left: 16px; right: 16px; bottom: 16px; pointer-events: none;"
  />
  <div
    class="ant-layout ant-layout-has-sider css-dev-only-do-not-override-98ntnt"
    style="min-height: 100vh;"
  >
    <aside
      class="ant-layout-sider ant-layout-sider-light sidebarContainer"
      style="flex: 0 0 170px; max-width: 170px; min-width: 170px; width: 170px;"
    >
      <div
        class="ant-layout-sider-children"
      >
        <div
          class="logoContainer ant-flex css-dev-only-do-not-override-98ntnt ant-flex-align-center ant-flex-justify-center"
          style="gap: 8px;"
        >
          <img
            alt="Litegraph logo"
            data-nimg="1"
            decoding="async"
            height="30"
            loading="lazy"
            src="/_next/image?url=%2Ffavicon.png&w=64&q=75"
            srcset="/_next/image?url=%2Ffavicon.png&w=32&q=75 1x, /_next/image?url=%2Ffavicon.png&w=64&q=75 2x"
            style="color: transparent;"
            width="30"
          />
          <h4
            class="ant-typography headingCommonStyles mt-xs css-dev-only-do-not-override-98ntnt"
            style="font-weight: 600;"
          >
            LiteGraph
          </h4>
        </div>
        <div
          class="pl-sm pr-sm pt-sm ant-flex css-dev-only-do-not-override-98ntnt ant-flex-justify-flex-end"
        >
          <button
            class="ant-btn css-dev-only-do-not-override-98ntnt ant-btn-link ant-btn-icon-only"
            style="font-size: 16px; right: 8%;"
            type="button"
          >
            <span
              class="ant-btn-icon"
            >
              <span
                aria-label="double-left"
                class="anticon anticon-double-left"
                role="img"
              >
                <svg
                  aria-hidden="true"
                  data-icon="double-left"
                  fill="currentColor"
                  focusable="false"
                  height="1em"
                  viewBox="64 64 896 896"
                  width="1em"
                >
                  <path
                    d="M272.9 512l265.4-339.1c4.1-5.2.4-12.9-6.3-12.9h-77.3c-4.9 0-9.6 2.3-12.6 6.1L186.8 492.3a31.99 31.99 0 000 39.5l255.3 326.1c3 3.9 7.7 6.1 12.6 6.1H532c6.7 0 10.4-7.7 6.3-12.9L272.9 512zm304 0l265.4-339.1c4.1-5.2.4-12.9-6.3-12.9h-77.3c-4.9 0-9.6 2.3-12.6 6.1L490.8 492.3a31.99 31.99 0 000 39.5l255.3 326.1c3 3.9 7.7 6.1 12.6 6.1H836c6.7 0 10.4-7.7 6.3-12.9L576.9 512z"
                  />
                </svg>
              </span>
            </span>
          </button>
        </div>
        <ul
          class="ant-menu ant-menu-root ant-menu-inline ant-menu-light css-dev-only-do-not-override-98ntnt"
          data-menu-list="true"
          role="menu"
          tabindex="0"
        >
          <li
            class="ant-menu-item"
            data-menu-id="rc-menu-uuid-test-/"
            role="menuitem"
            style="padding-left: 24px;"
            tabindex="-1"
          >
            <span
              aria-label="home"
              class="anticon anticon-home ant-menu-item-icon"
              role="img"
            >
              <svg
                aria-hidden="true"
                data-icon="home"
                fill="currentColor"
                focusable="false"
                height="1em"
                viewBox="64 64 896 896"
                width="1em"
              >
                <path
                  d="M946.5 505L560.1 118.8l-25.9-25.9a31.5 31.5 0 00-44.4 0L77.5 505a63.9 63.9 0 00-18.8 46c.4 35.2 29.7 63.3 64.9 63.3h42.5V940h691.8V614.3h43.4c17.1 0 33.2-6.7 45.3-18.8a63.6 63.6 0 0018.7-45.3c0-17-6.7-33.1-18.8-45.2zM568 868H456V664h112v204zm217.9-325.7V868H632V640c0-22.1-17.9-40-40-40H432c-22.1 0-40 17.9-40 40v228H238.1V542.3h-96l370-369.7 23.1 23.1L882 542.3h-96.1z"
                />
              </svg>
            </span>
            <span
              class="ant-menu-title-content"
            >
              <a
                href="/dashboard"
              >
                <span>
                  Home
                </span>
              </a>
            </span>
          </li>
          <li
            class="ant-menu-item"
            data-menu-id="rc-menu-uuid-test-/graphs"
            role="menuitem"
            style="padding-left: 24px;"
            tabindex="-1"
          >
            <span
              aria-label="share-alt"
              class="anticon anticon-share-alt ant-menu-item-icon"
              role="img"
            >
              <svg
                aria-hidden="true"
                data-icon="share-alt"
                fill="currentColor"
                focusable="false"
                height="1em"
                viewBox="64 64 896 896"
                width="1em"
              >
                <path
                  d="M752 664c-28.5 0-54.8 10-75.4 26.7L469.4 540.8a160.68 160.68 0 000-57.6l207.2-149.9C697.2 350 723.5 360 752 360c66.2 0 120-53.8 120-120s-53.8-120-120-120-120 53.8-120 120c0 11.6 1.6 22.7 4.7 33.3L439.9 415.8C410.7 377.1 364.3 352 312 352c-88.4 0-160 71.6-160 160s71.6 160 160 160c52.3 0 98.7-25.1 127.9-63.8l196.8 142.5c-3.1 10.6-4.7 21.8-4.7 33.3 0 66.2 53.8 120 120 120s120-53.8 120-120-53.8-120-120-120zm0-476c28.7 0 52 23.3 52 52s-23.3 52-52 52-52-23.3-52-52 23.3-52 52-52zM312 600c-48.5 0-88-39.5-88-88s39.5-88 88-88 88 39.5 88 88-39.5 88-88 88zm440 236c-28.7 0-52-23.3-52-52s23.3-52 52-52 52 23.3 52 52-23.3 52-52 52z"
                />
              </svg>
            </span>
            <span
              class="ant-menu-title-content"
            >
              <a
                href="/dashboard//graphs"
              >
                <span>
                  Graphs
                </span>
              </a>
            </span>
          </li>
          <li
            class="ant-menu-item"
            data-menu-id="rc-menu-uuid-test-/nodes"
            role="menuitem"
            style="padding-left: 24px;"
            tabindex="-1"
          >
            <span
              aria-label="node-index"
              class="anticon anticon-node-index ant-menu-item-icon"
              role="img"
            >
              <svg
                aria-hidden="true"
                data-icon="node-index"
                fill="currentColor"
                focusable="false"
                height="1em"
                viewBox="64 64 896 896"
                width="1em"
              >
                <defs>
                  <style />
                </defs>
                <path
                  d="M843.5 737.4c-12.4-75.2-79.2-129.1-155.3-125.4S550.9 676 546 752c-153.5-4.8-208-40.7-199.1-113.7 3.3-27.3 19.8-41.9 50.1-49 18.4-4.3 38.8-4.9 57.3-3.2 1.7.2 3.5.3 5.2.5 11.3 2.7 22.8 5 34.3 6.8 34.1 5.6 68.8 8.4 101.8 6.6 92.8-5 156-45.9 159.2-132.7 3.1-84.1-54.7-143.7-147.9-183.6-29.9-12.8-61.6-22.7-93.3-30.2-14.3-3.4-26.3-5.7-35.2-7.2-7.9-75.9-71.5-133.8-147.8-134.4-76.3-.6-140.9 56.1-150.1 131.9s40 146.3 114.2 163.9c74.2 17.6 149.9-23.3 175.7-95.1 9.4 1.7 18.7 3.6 28 5.8 28.2 6.6 56.4 15.4 82.4 26.6 70.7 30.2 109.3 70.1 107.5 119.9-1.6 44.6-33.6 65.2-96.2 68.6-27.5 1.5-57.6-.9-87.3-5.8-8.3-1.4-15.9-2.8-22.6-4.3-3.9-.8-6.6-1.5-7.8-1.8l-3.1-.6c-2.2-.3-5.9-.8-10.7-1.3-25-2.3-52.1-1.5-78.5 4.6-55.2 12.9-93.9 47.2-101.1 105.8-15.7 126.2 78.6 184.7 276 188.9 29.1 70.4 106.4 107.9 179.6 87 73.3-20.9 119.3-93.4 106.9-168.6zM329.1 345.2a83.3 83.3 0 11.01-166.61 83.3 83.3 0 01-.01 166.61zM695.6 845a83.3 83.3 0 11.01-166.61A83.3 83.3 0 01695.6 845z"
                />
              </svg>
            </span>
            <span
              class="ant-menu-title-content"
            >
              <a
                href="/dashboard//nodes"
              >
                <span>
                  Nodes
                </span>
              </a>
            </span>
          </li>
          <li
            class="ant-menu-item"
            data-menu-id="rc-menu-uuid-test-/edges"
            role="menuitem"
            style="padding-left: 24px;"
            tabindex="-1"
          >
            <span
              aria-label="api"
              class="anticon anticon-api ant-menu-item-icon"
              role="img"
            >
              <svg
                aria-hidden="true"
                data-icon="api"
                fill="currentColor"
                focusable="false"
                height="1em"
                viewBox="64 64 896 896"
                width="1em"
              >
                <path
                  d="M917.7 148.8l-42.4-42.4c-1.6-1.6-3.6-2.3-5.7-2.3s-4.1.8-5.7 2.3l-76.1 76.1a199.27 199.27 0 00-112.1-34.3c-51.2 0-102.4 19.5-141.5 58.6L432.3 308.7a8.03 8.03 0 000 11.3L704 591.7c1.6 1.6 3.6 2.3 5.7 2.3 2 0 4.1-.8 5.7-2.3l101.9-101.9c68.9-69 77-175.7 24.3-253.5l76.1-76.1c3.1-3.2 3.1-8.3 0-11.4zM769.1 441.7l-59.4 59.4-186.8-186.8 59.4-59.4c24.9-24.9 58.1-38.7 93.4-38.7 35.3 0 68.4 13.7 93.4 38.7 24.9 24.9 38.7 58.1 38.7 93.4 0 35.3-13.8 68.4-38.7 93.4zm-190.2 105a8.03 8.03 0 00-11.3 0L501 613.3 410.7 523l66.7-66.7c3.1-3.1 3.1-8.2 0-11.3L441 408.6a8.03 8.03 0 00-11.3 0L363 475.3l-43-43a7.85 7.85 0 00-5.7-2.3c-2 0-4.1.8-5.7 2.3L206.8 534.2c-68.9 69-77 175.7-24.3 253.5l-76.1 76.1a8.03 8.03 0 000 11.3l42.4 42.4c1.6 1.6 3.6 2.3 5.7 2.3s4.1-.8 5.7-2.3l76.1-76.1c33.7 22.9 72.9 34.3 112.1 34.3 51.2 0 102.4-19.5 141.5-58.6l101.9-101.9c3.1-3.1 3.1-8.2 0-11.3l-43-43 66.7-66.7c3.1-3.1 3.1-8.2 0-11.3l-36.6-36.2zM441.7 769.1a131.32 131.32 0 01-93.4 38.7c-35.3 0-68.4-13.7-93.4-38.7a131.32 131.32 0 01-38.7-93.4c0-35.3 13.7-68.4 38.7-93.4l59.4-59.4 186.8 186.8-59.4 59.4z"
                />
              </svg>
            </span>
            <span
              class="ant-menu-title-content"
            >
              <a
                href="/dashboard//edges"
              >
                <span>
                  Edges
                </span>
              </a>
            </span>
          </li>
          <li
            class="ant-menu-item"
            data-menu-id="rc-menu-uuid-test-/labels"
            role="menuitem"
            style="padding-left: 24px;"
            tabindex="-1"
          >
            <span
              aria-label="tags"
              class="anticon anticon-tags ant-menu-item-icon"
              role="img"
            >
              <svg
                aria-hidden="true"
                data-icon="tags"
                fill="currentColor"
                focusable="false"
                height="1em"
                viewBox="64 64 896 896"
                width="1em"
              >
                <path
                  d="M483.2 790.3L861.4 412c1.7-1.7 2.5-4 2.3-6.3l-25.5-301.4c-.7-7.8-6.8-13.9-14.6-14.6L522.2 64.3c-2.3-.2-4.7.6-6.3 2.3L137.7 444.8a8.03 8.03 0 000 11.3l334.2 334.2c3.1 3.2 8.2 3.2 11.3 0zm62.6-651.7l224.6 19 19 224.6L477.5 694 233.9 450.5l311.9-311.9zm60.16 186.23a48 48 0 1067.88-67.89 48 48 0 10-67.88 67.89zM889.7 539.8l-39.6-39.5a8.03 8.03 0 00-11.3 0l-362 361.3-237.6-237a8.03 8.03 0 00-11.3 0l-39.6 39.5a8.03 8.03 0 000 11.3l243.2 242.8 39.6 39.5c3.1 3.1 8.2 3.1 11.3 0l407.3-406.6c3.1-3.1 3.1-8.2 0-11.3z"
                />
              </svg>
            </span>
            <span
              class="ant-menu-title-content"
            >
              <a
                href="/dashboard//labels"
              >
                <span>
                  Labels
                </span>
              </a>
            </span>
          </li>
          <li
            class="ant-menu-item"
            data-menu-id="rc-menu-uuid-test-/tags"
            role="menuitem"
            style="padding-left: 24px;"
            tabindex="-1"
          >
            <span
              aria-label="tags"
              class="anticon anticon-tags ant-menu-item-icon"
              role="img"
            >
              <svg
                aria-hidden="true"
                data-icon="tags"
                fill="currentColor"
                focusable="false"
                height="1em"
                viewBox="64 64 896 896"
                width="1em"
              >
                <path
                  d="M483.2 790.3L861.4 412c1.7-1.7 2.5-4 2.3-6.3l-25.5-301.4c-.7-7.8-6.8-13.9-14.6-14.6L522.2 64.3c-2.3-.2-4.7.6-6.3 2.3L137.7 444.8a8.03 8.03 0 000 11.3l334.2 334.2c3.1 3.2 8.2 3.2 11.3 0zm62.6-651.7l224.6 19 19 224.6L477.5 694 233.9 450.5l311.9-311.9zm60.16 186.23a48 48 0 1067.88-67.89 48 48 0 10-67.88 67.89zM889.7 539.8l-39.6-39.5a8.03 8.03 0 00-11.3 0l-362 361.3-237.6-237a8.03 8.03 0 00-11.3 0l-39.6 39.5a8.03 8.03 0 000 11.3l243.2 242.8 39.6 39.5c3.1 3.1 8.2 3.1 11.3 0l407.3-406.6c3.1-3.1 3.1-8.2 0-11.3z"
                />
              </svg>
            </span>
            <span
              class="ant-menu-title-content"
            >
              <a
                href="/dashboard//tags"
              >
                <span>
                  Tags
                </span>
              </a>
            </span>
          </li>
          <li
            class="ant-menu-item"
            data-menu-id="rc-menu-uuid-test-/vectors"
            role="menuitem"
            style="padding-left: 24px;"
            tabindex="-1"
          >
            <span
              aria-label="tags"
              class="anticon anticon-tags ant-menu-item-icon"
              role="img"
            >
              <svg
                aria-hidden="true"
                data-icon="tags"
                fill="currentColor"
                focusable="false"
                height="1em"
                viewBox="64 64 896 896"
                width="1em"
              >
                <path
                  d="M483.2 790.3L861.4 412c1.7-1.7 2.5-4 2.3-6.3l-25.5-301.4c-.7-7.8-6.8-13.9-14.6-14.6L522.2 64.3c-2.3-.2-4.7.6-6.3 2.3L137.7 444.8a8.03 8.03 0 000 11.3l334.2 334.2c3.1 3.2 8.2 3.2 11.3 0zm62.6-651.7l224.6 19 19 224.6L477.5 694 233.9 450.5l311.9-311.9zm60.16 186.23a48 48 0 1067.88-67.89 48 48 0 10-67.88 67.89zM889.7 539.8l-39.6-39.5a8.03 8.03 0 00-11.3 0l-362 361.3-237.6-237a8.03 8.03 0 00-11.3 0l-39.6 39.5a8.03 8.03 0 000 11.3l243.2 242.8 39.6 39.5c3.1 3.1 8.2 3.1 11.3 0l407.3-406.6c3.1-3.1 3.1-8.2 0-11.3z"
                />
              </svg>
            </span>
            <span
              class="ant-menu-title-content"
            >
              <a
                href="/dashboard//vectors"
              >
                <span>
                  Vectors
                </span>
              </a>
            </span>
          </li>
        </ul>
        <div
          aria-hidden="true"
          style="display: none;"
        />
      </div>
    </aside>
    <div
      class="ant-layout css-dev-only-do-not-override-98ntnt"
    >
      <header
        class="ant-layout-header header css-dev-only-do-not-override-98ntnt"
      >
        <div
          class="ant-flex css-dev-only-do-not-override-98ntnt ant-flex-align-stretch ant-flex-justify-center ant-flex-vertical"
        >
          <div
            class="ant-flex css-dev-only-do-not-override-98ntnt ant-flex-align-center"
            style="gap: 8px;"
          >
            <span>
              Graph:
            </span>
            <select
              data-testid="litegraph-select"
            />
          </div>
        </div>
        <div
          class="userSection"
        >
          <div
            class="ant-dropdown-trigger container ant-flex css-dev-only-do-not-override-98ntnt ant-flex-align-center"
            style="gap: 10px;"
          >
            <span
              class="ant-typography ant-color-white css-dev-only-do-not-override-98ntnt"
              style="font-weight: 400;"
            >
              <strong>
                User
              </strong>
            </span>
            <span
              class="ant-avatar ant-avatar-sm ant-avatar-circle css-dev-only-do-not-override-98ntnt"
            >
              <span
                class="ant-avatar-string"
                style="transform: scale(1);"
              >
                U
              </span>
            </span>
            <span
              aria-label="down-circle"
              class="anticon anticon-down-circle ant-color-white"
              role="img"
            >
              <svg
                aria-hidden="true"
                data-icon="down-circle"
                fill="currentColor"
                focusable="false"
                height="1em"
                viewBox="64 64 896 896"
                width="1em"
              >
                <path
                  d="M690 405h-46.9c-10.2 0-19.9 4.9-25.9 13.2L512 563.6 406.8 418.2c-6-8.3-15.6-13.2-25.9-13.2H334c-6.5 0-10.3 7.4-6.5 12.7l178 246c3.2 4.4 9.7 4.4 12.9 0l178-246c3.9-5.3.1-12.7-6.4-12.7z"
                />
                <path
                  d="M512 64C264.6 64 64 264.6 64 512s200.6 448 448 448 448-200.6 448-448S759.4 64 512 64zm0 820c-205.4 0-372-166.6-372-372s166.6-372 372-372 372 166.6 372 372-166.6 372-372 372z"
                />
              </svg>
            </span>
          </div>
        </div>
      </header>
      <main
        class="ant-layout-content css-dev-only-do-not-override-98ntnt"
        style="min-height: 280px; background: rgb(255, 255, 255);"
      >
        <main
          class="ant-layout-content pageContainer css-dev-only-do-not-override-98ntnt"
        >
          <div
            class="pageContent"
          >
            <div
              class="ant-flex css-dev-only-do-not-override-98ntnt ant-flex-align-center ant-flex-justify-center ant-flex-vertical"
            >
              <span
                class="ant-typography css-dev-only-do-not-override-98ntnt"
              >
                Loading...
              </span>
              <span
                aria-label="loading"
                class="anticon anticon-loading anticon-spin pageLoader"
                role="img"
              >
                <svg
                  aria-hidden="true"
                  data-icon="loading"
                  fill="currentColor"
                  focusable="false"
                  height="1em"
                  viewBox="0 0 1024 1024"
                  width="1em"
                >
                  <path
                    d="M988 548c-19.9 0-36-16.1-36-36 0-59.4-11.6-117-34.6-171.3a440.45 440.45 0 00-94.3-139.9 437.71 437.71 0 00-139.9-94.3C629 83.6 571.4 72 512 72c-19.9 0-36-16.1-36-36s16.1-36 36-36c69.1 0 136.2 13.5 199.3 40.3C772.3 66 827 103 874 150c47 47 83.9 101.8 109.7 162.7 26.7 63.1 40.2 130.2 40.2 199.3.1 19.9-16 36-35.9 36z"
                  />
                </svg>
              </span>
            </div>
          </div>
        </main>
      </main>
    </div>
  </div>
</div>
`;
